### PR TITLE
DP-3226 a11y Heading or Label is Not Descriptive - Body

### PR DESCRIPTION
**Pull request by ygannett**
## Description
Add context with page titles or section titles in `span.visually-hidden` to preset heading titles (not created by authors) repeatedly used through the sites such as "Overview", "Contacts", "Downloads", "Parking" and etc. for AT users.  

Such supplemental information is not visible to sighted users in a browser.  It is only recognized by AT devices such as screen readers.

### Affected page types

- Location
- Advisory
- Decision
- Event
- Executive order
- Guide
- How-to
- Location Detail
- Service Detail
- News
- Organization
- Regulation 
- Service

## Related Issue / Ticket

- [DP-3226 a11y Heading or Label is Not Descriptive - Body](https://jira.state.ma.us/browse/DP-3226)

## Steps to Test

1. Open above pages of the page types listed above in a browser.
2. Open the inspector or open the page source.
3. Locate preset headings commonly used in those page types in the source.
4. Find the headings have supplemental information with the page title or section title in `span.visually-hidden` like the sample.
pattern_lab_-_pages-location-general-content_and_by-author
Mostly, their page titles are used for the supplemental information. The guide page uses each section unit title for sections, but the related guide uses the page title.
Guide - Section
pattern_lab_-_pages-guide_and_by-author
Guide - Related Guide
guide-related

## Screenshots
No visual change in a browser.

## Additional Notes:

This change has corresponding PR for Drupal:
[DP-3226 A11y Heading or Label is Not Descriptive - Body #1348](https://github.com/massgov/mass/pull/1348)

#### Impacted Areas in Application

* 

#### @TODO

*

### CODE REVIEW

File: styleguide/source/_patterns/02-molecules/sticky-nav.twig [hunk 1]
```diff
@@ -1,6 +1,7 @@
 <nav 
   class="ma__sticky-nav js-scroll-anchors" 
   aria-label="Page Contents">
+  <h3 class="visually-hidden">the Contents of the {{stickyNav.titleContext}} page</h3>
```
Comment (powpow12) [inline styleguide/source/_patterns/02-molecules/sticky-nav.twig]: this forgos the logic we've seen elsewhere that checks if titleContext exists : e.g,
```
{% if sidebarHeading.titleContext %} +  <span class="visually-hidden"> {{sidebarHeading.titleContext}}</span>{% endif %}
```

File: styleguide/source/_patterns/03-organisms/by-author/action-details.twig [hunk 2]
```diff
@@ -1,9 +1,9 @@
 <section class="ma__action-details">
-  <h2 class="ma__action-details__title">The Details</h2>
+  <h2 class="ma__action-details__title">The Details{% if actionDetails.titleContext %}<span class="visually-hidden"> of {{actionDetails.titleContext}}</span>{% endif %}</h2>
 
   <div class="ma__action-details__content">
     <nav class="ma__action-details__anchor-links js-scroll-anchors" aria-labelledby="page_contents">
-        <h3 class="visually-hidden" id="page_contents">Page Contents</h3>
+        <h3 class="visually-hidden" id="page_contents">Page Contents of {{actionDetails.titleContext}}</h3>
```
Comment (powpow12) [inline styleguide/source/_patterns/03-organisms/by-author/action-details.twig]: missing if value exist check

**Review by powpow12 — DISMISSED** [review]

File: styleguide/source/_patterns/02-molecules/sticky-nav.twig (hunk 1 above)
Comment (ygannett) [reply styleguide/source/_patterns/02-molecules/sticky-nav.twig]: @powpow12 Thank you for pointing that out.  I must have missed it.  The value check condition has been added to this and another one you pointed.

**Review by isaacchansky — APPROVED** [review]
This looks good to me. @powpow12 can you take another look?

**Pull request merged**